### PR TITLE
tests: reduce duplication of upgrade tests + proxy backcompat tests

### PR DESCRIPTION
**Pull request by orndorffgrant**
This tries to reduce the complexity of the upgrade tests to avoid problems we ran into last release, and hopefully make adding new ubuntu releases easier in the future.

It also adjusts the warning message for the old proxy config field slightly and adds a new integration test for the backwards compatibility to use the old config field.

## Test Steps

Run the new integration tests

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/cli.py [hunk 1]
```diff
@@ -901,14 +901,19 @@ def action_config_set(args, *, cfg, **kwargs):
...
+        protocol_type = "https" if "https" in set_key else "http"
         if protocol_type == "http":
             validate_url = util.PROXY_VALIDATION_APT_HTTP_URL
+            s = ""
         else:
             validate_url = util.PROXY_VALIDATION_APT_HTTPS_URL
+            s = "s"
+
+        if set_key in cfg.deprecated_global_scoped_proxy_options:
+            print(messages.WARNING_APT_PROXY_SETUP.format(s=s))
```
Comment (lucasmoura) [inline uaclient/cli.py:914]: I think we can drop the `s` variable and just pass the `protocol_ type` directly to the string right ?

We would not need to turn it into:
`Warning: apt_{protocol_type}_proxy has been renamed to global_apt_{protocol_type}_proxy`

Comment (orndorffgrant) [reply uaclient/cli.py:914]: yep thanks! thats much simpler, I'll make that change

**Comment by lucasmoura** [comment]
@orndorffgrant great refactoring and all tests passed for me. After you address the string change and @renanrodrigo comment, we can merge this

File: uaclient/cli.py (hunk 1 above)
Comment (orndorffgrant) [reply uaclient/cli.py]: updated and pushed!

File: features/ubuntu_upgrade_unattached.feature [hunk 2]
```diff
@@ -2,52 +2,10 @@
...
-        | impish   | jammy        |
-
-   @slow
-   @series.xenial
-   @series.bionic
-   @series.focal
-   @uses.config.machine_type.lxd.container
-   @upgrade
-   Scenario Outline: Unattached upgrade across LTS releases
+    Scenario Outline: Attached upgrade
```
[the remark this answers is not in the document]
Comment (orndorffgrant) [reply features/ubuntu_upgrade_unattached.feature]: updated and pushed!

**Review by lucasmoura — APPROVED** [review]

**Review by renanrodrigo — APPROVED** [review]

**Pull request merged**